modules/standard/IO.chpl

```diff
@@ -6643,17 +6603,25 @@ proc channel.readf(fmtStr:?t, ref args ...?k): bool throws
         // revert
         qio_channel_revert_unlocked(_channel_internal);
       }
+    } catch thrownError: SystemError {
+      if thrownError.err != EEOF then throw thrownError;
+      err = EEOF;
     }
-    this._set_style(save_style);
   }
 
   if !err {
     return true;
   } else if err == EEOF {
+    //
+    // TODO: Shouldn't we also return false for EFORMAT? Yet some tests
+    // (io/ferguson/json.chpl off the top of my head) expect EFORMAT errors
+    // to be thrown. Is the documentation off?
```

Comment:
What made you think it should return false for EFORMAT? I think I'd expect it to throw.

Reply:
The documentation for `channel.readf` says that it "false on EEOF or if the format did not match the input". Should the docstring be updated instead?

Reply:
I don't know what to do with it. Maybe make an issue about it?

Reply:
Sure, I can do that.